pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -743,10 +746,45 @@ func (iqr *IQR) RenameColumn(oldName, newName string) error {
 		return err
 	}
 
-	iqr.renamedColumns[oldName] = newName
-	if values, ok := iqr.knownValues[oldName]; ok {
+	// delete newName since it would be overwritten
+	delete(iqr.knownValues, newName)
+
+	values, ok := iqr.knownValues[oldName]
+	if ok {
+		// if old name is present it means that it must be a created column
+		// so we will rename it and update the knownValues map
+		iqr.renamedColumns[oldName] = newName
 		iqr.knownValues[newName] = values
-		delete(iqr.knownValues, oldName)
```

Comment:
> if old name is present it means that it must be a created column so we will rename it and update the knownValues map

I don't think that's true. I think when we read RRCs for a certain column, we also cache that in `knownValues`. Maybe I misunderstand what you mean by "created column"

Reply:
But reading the column is not necessary for all the commands. 
For example consider `* | rename first_name as firstName`, here `first_name` would never be read because its not required you just need to mark it for renaming. And there might be more types of commands where there is a similar case.

Reply:
Oh I forgot to mention that cached columns can also be present over here. But, I think this is fine, since caching would probably be made optional and we would not want to mark this as a `deletedColumn` because we would need to read it for the case mentioned above.